### DISCUSSION TITLE
Transformer Source Code: Example & Question About  a Repeating Item

### DISCUSSION THREAD

**Discussion by flugruger** [nextgenhealthcare/connect#5021] [Q&A] [answered]
In [another thread,](https://github.com/nextgenhealthcare/connect/discussions/4847#discussioncomment-1738296) @odoodo provided a way to export the source code of a transformer. This post follows on that with two items:

- An example of Mirth's core destination transformer code ([Core Transformer v2.js.txt](https://github.com/nextgenhealthcare/connect/files/8056046/Core.Transformer.v2.js.txt));
- ~~A question about one part of the output.~~

Edit: Looked more closely. No longer a question. The first if else if is for the msg, the second is for the tmp. 
Edit 2: Formatting, per @pacmano1. 

In the attached file, the core of the transformer is this JS: 

```js
function doScript() {
msg = new XML(connectorMessage.getTransformedData());
if (msg.namespace('') != undefined) { default xml namespace = msg.namespace(''); } else { default xml namespace = ''; }
tmp = JSON.parse(template);
function filterRule1() {

	// FILTER JS WAS HERE, REMOVED
}
}function doFilter() { phase[0] = 'filter'; return ((filterRule1() == true)); }
function doTransform() { phase[0] = 'transformer'; logger = Packages.org.apache.log4j.Logger.getLogger(phase[0]);

	// TRANSFORM STEPS WERE HERE, JS REMOVED

if ('xml' === typeof msg) {
    if (msg.hasSimpleContent()) {
        msg = msg.toXMLString();
    }
} else if ('undefined' !== typeof msg && msg !== null) {
    var toStringResult = Object.prototype.toString.call(msg);
    if (toStringResult == '[object Object]' || toStringResult == '[object Array]') {
        msg = JSON.stringify(msg);
    }
}
if ('xml' === typeof tmp) {
    if (tmp.hasSimpleContent()) {
        tmp = tmp.toXMLString();
    }
} else if ('undefined' !== typeof tmp && tmp !== null) {
    var toStringResult = Object.prototype.toString.call(tmp);
    if (toStringResult == '[object Object]' || toStringResult == '[object Array]') {
        tmp = JSON.stringify(tmp);
    }
}
}
if (doFilter() == true) { doTransform(); return true; } else { return false; } 
}
doScript();
```

My question is about the if else if statements below where the transformer code was. 
Those seem to be duplicates. Is this an artifact or is there some reason for doing this?

Thanks!

**Comment by pacmano1** [comment, marked as answer]
@flugruger use code highlighting please, you know better :).

https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks

https://github.com/nextgenhealthcare/connect/discussions/4824

**Reply by flugruger** [reply]
As usual, @pacmano1 , I learned something from you. I had hit the code button in the editor and couldn't understand why it wasn't doing a code block. Now I know that I have to hit it three times. :)

**Reply by pacmano1** [reply]
if you add js after the 3 back ticks it colors the code also.

```js
function doScript() {
    msg = new XML(connectorMessage.getTransformedData());
}
```

Raw viiew of that:

![image](https://user-images.githubusercontent.com/44065187/153766259-247ca908-e9de-4a14-81a6-92d0aa5a067a.png)

**Reply by flugruger** [reply]
Thanks @pacmano1 !

**Comment by tonygermano** [comment]
They are not duplicates. The first if/else statement operates on the `msg` variable, and the second operates on the `tmp` variable.
